front/temporal/labs/utils/gong.ts

```diff
@@ -125,14 +159,72 @@ export async function retrieveGongTranscriptContent(
       {},
       "[processTranscriptActivity] Call data not found from Gong. Skipping."
     );
-    throw new Error("Call data not found from Gong. Skipping.");
+    return null;
   }
 
-  const participants: { [key: string]: string } = {};
-  callData.participants?.map((participant: GongParticipant) => {
-    participants[participant.speakerId] = participant.name;
+  const user = await transcriptsConfiguration.getUser();
+
+  if (!user) {
+    localLogger.error(
+      {},
+      "[processTranscriptActivity] User not found. Skipping."
+    );
+    return null;
+  }
+
+  const gongUsers = await fetch(`https://api.gong.io/v2/users`, {
+    headers: {
+      Authorization: `Bearer ${gongAccessToken}`,
+      "Content-Type": "application/json",
+    },
+  });
+
+  if (!gongUsers.ok) {
+    localLogger.error(
+      {},
+      "[retrieveGongTranscripts] Error fetching Gong users. Skipping."
+    );
+    return null;
+  }
+
+  const gongUsersData = await gongUsers.json();
+
+  if (!gongUsersData || gongUsersData.length === 0) {
+    localLogger.warn(
+      {},
+      "[retrieveGongTranscripts] No Gong users found. Skipping."
+    );
+    return null;
+  }
+
+  const gongUser = gongUsersData.users.find(
+    (gongUser: { emailAddress: string }) => gongUser.emailAddress === user.email
+  );
+
+  if (!gongUser) {
+    localLogger.warn(
+      {},
+      "[retrieveGongTranscripts] User not found in Gong. Skipping."
+    );
+    return null;
+  }
+
+  const participantsUsers: { [key: string]: string } = {};
+  const participantsSpeakers: { [key: string]: string } = {};
+
+  callData.parties?.map((participant: GongParticipant) => {
```

Comment:
Can we do a proper map implementation here? Since this map mutates a local variable. Or you can do a reduce.

Reply:
Switched to a for loop - reduce implied doing 2 separate loops with pretty much the same code and I didn't like it :')